RecoHGCal/TICL/plugins/PatternRecognitionbyCA.cc

```diff
@@ -63,22 +58,32 @@ void PatternRecognitionbyCA::makeTracksters(const edm::Event &ev, const edm::Eve
       effective_cluster_idx.insert(innerCluster);
       effective_cluster_idx.insert(outerCluster);
       if (algo_verbosity_ > Advanced) {
-        LogDebug("HGCPatterRecoByCA")
-            << "New doublet " << doublet << " for trackster: " << result.size() << " InnerCl "
-            << innerCluster << " " << layerClusters[innerCluster].x() << " "
-            << layerClusters[innerCluster].y() << " " << layerClusters[innerCluster].z()
-            << " OuterCl " << outerCluster << " " << layerClusters[outerCluster].x() << " "
-            << layerClusters[outerCluster].y() << " " << layerClusters[outerCluster].z() << " "
-            << tracksterId << std::endl;
+        LogDebug("HGCPatterRecoByCA") << "New doublet " << doublet << " for trackster: " << result.size() << " InnerCl "
+                                      << innerCluster << " " << layerClusters[innerCluster].x() << " "
+                                      << layerClusters[innerCluster].y() << " " << layerClusters[innerCluster].z()
+                                      << " OuterCl " << outerCluster << " " << layerClusters[outerCluster].x() << " "
+                                      << layerClusters[outerCluster].y() << " " << layerClusters[outerCluster].z()
+                                      << " " << tracksterId << std::endl;
       }
     }
+    for (auto const i : effective_cluster_idx) {
+      layer_cluster_usage[i]++;
+      LogDebug("HGCPatterRecoByCA") << "LayerID: " << i << " count: " << (int)layer_cluster_usage[i] << std::endl;
+    }
     // Put back indices, in the form of a Trackster, into the results vector
     Trackster tmp;
     tmp.vertices.reserve(effective_cluster_idx.size());
-    std::copy(std::begin(effective_cluster_idx), std::end(effective_cluster_idx),
-              std::back_inserter(tmp.vertices));
+    tmp.vertex_multiplicity.resize(effective_cluster_idx.size(), 0);
+    std::copy(std::begin(effective_cluster_idx), std::end(effective_cluster_idx), std::back_inserter(tmp.vertices));
```

Comment:
I think this would be simpler (and maybe better performing) as:
```
tmp.vertices.insert(tmp.vertices.end(),effective_cluster_idx.begin(),effective_cluster_idx.end());
```